### ISSUE TITLE
Install `helmet` for non-`ltijs` routes

### ISSUE THREAD

**Issue by ssciolla** [tl-its-umich-edu/canvas-course-manager-next#166]
As part of PR #148 (issue #139) or after it is merged, we should install `helmet` across the application, as I don't believe it will be applied to non-`ltijs` routes (`ltijs` uses `helmet`). See the links below for info about what `helmet` does and how to implement it.

Link(s):
- https://github.com/helmetjs/helmet
- http://expressjs.com/en/advanced/best-practice-security.html#use-helmet
- https://docs.nestjs.com/security/helmet#helmet

**Comment by pushyamig** [comment]
@ssciolla how do I test this?

**Comment by pushyamig** [comment]
@ssciolla How do I test this?

**Comment by ssciolla** [comment]
So, I looked into this a bit. Here are two ways I would test it:

1) Add logging around the middleware, something like this:
In `app.module.ts`
```
    consumer.apply((req: Request, res: Response, next: NextFunction) => {
      const helmetMid = helmet({
        contentSecurityPolicy: {
          directives: { 'frame-ancestors': [this.frameDomain] }
        }
      })
      console.log('before helmet')
      helmetMid(req, res, next)
      console.log('after helmet')
    })
```
`Request`, `Response`, and `NextFunction` are imported from `express`. You'll see that non-`ltijs` calls will have log statements.

2) Look at the expected headers in this article, and check network calls in your browser to see whether the headers are there: https://www.twilio.com/blog/securing-your-express-app-html

**Comment by ssciolla** [comment]
Looking at this again, I'm not wondering whether we may need to install `helmet` as global middleware, rather than module testing. I did this originally because I didn't want to install `helmet` twice on the `ltijs` routes (I was able to exclude them with the module approach). What do you think? See post below.

https://stackoverflow.com/questions/41492312/middleware-order-express

**Comment by pushyamig** [comment]
I would say Let's hold on this. I don't see a huge impact in terms if performance for the app with way adding at global or module level. You can create an issue if you agree.

**Comment by pushyamig** [comment]
I will QA this

**Comment by pushyamig** [comment]
When I added the code as suggested the helmet was adding the below header only to the response object
`x-powered-by: Express`

But after I took the custom code out and back to the way it was I see following headers to the response. This is something helmet will add. 

```
x-content-type-options: nosniff
	x-dns-prefetch-control: off
	x-download-options: noopen
	x-frame-options: SAMEORIGIN
	x-permitted-cross-domain-policies: none
	x-xss-protection: 0
```

After reading the article it talks about this exact difference, on that note I will make this pass.
